src/PgRvn.Server/MessageBuilder.cs

```diff
@@ -0,0 +1,365 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PgRvn.Server
+{
+    public class MessageBuilder : IDisposable
+    {
+        private readonly IMemoryOwner<byte> _bufferOwner;
+        private Memory<byte> Buffer => _bufferOwner.Memory;
+
+        public MessageBuilder()
+        {
+            _bufferOwner = MemoryPool<byte>.Shared.Rent(32*1024); // TODO: need to manage sizes here
```

Comment:
How many of those are we likely to have? One per connection?

Reply:
What happens if we have a query > 32KB? Or a response that size?

Reply:
Added buffer size management